backend/src/main/java/gov/cdc/usds/simplereport/service/ApiUserService.java

```diff
@@ -664,4 +644,64 @@ public Set<String> getTenantDataAccessAuthoritiesForCurrentUser() {
     ApiUser apiUser = getCurrentApiUser();
     return _tenantService.getTenantDataAccessAuthorities(apiUser);
   }
+
+  @AuthorizationConfiguration.RequireGlobalAdminUser
+  public UserInfo getUserByLoginEmail(String loginEmail) {
+    Optional<ApiUser> foundUser = _apiUserRepo.findByLoginEmailIncludeArchived(loginEmail);
+
+    if (foundUser.isEmpty()) {
+      throw new NonexistentUserException();
+    }
+
+    ApiUser apiUser = foundUser.get();
+
+    try {
+      // We hit the okta API just once and resolve the info of status, isAdmin and organization
+      // roles with the same user object. It was not possible to move the entire
+      // consolidateUserInformation logic in the okta repo because the reference to the orgService
+      // (necessary to create the org roles) creates a cycle dependency
+      PartialOktaUser oktaUser = _oktaRepo.findUser(apiUser.getLoginEmail());
+      boolean isAdmin = oktaUser.isAdmin();
+
+      if (isAdmin) {
+        throw new RestrictedAccessUserException();
+      }
+
+      return consolidateUser(
+          apiUser, oktaUser.getOrganizationRoleClaims(), oktaUser.getStatus(), oktaUser.isAdmin());
+    } catch (IllegalGraphqlArgumentException | UnidentifiedUserException e) {
+      throw new OktaAccountUserException();
+    }
+  }
+
+  private UserInfo consolidateUser(
+      ApiUser apiUser,
+      Optional<OrganizationRoleClaims> optClaims,
+      UserStatus userStatus,
+      Boolean isAdmin) {
+    if (optClaims.isEmpty()) {
+      throw new UnidentifiedUserException();
+    }
+
+    OrganizationRoleClaims claims = optClaims.get();
```

Comment:
nit: If you plan on already making changes you can use the [`.orElseThrow()` method on `Optional`](https://www.baeldung.com/java-optional-throw-exception#optionalorelsethrowmethod), this will clean this up a little bit
```java
    OrganizationRoleClaims claims = optClaims.orElseThrow(UnidentifiedUserException::new);
```